src/main/kotlin/org/rust/cargo/project/CargoProjectDescription.kt

```diff
@@ -102,7 +100,7 @@ class CargoProjectDescription private constructor(
 
     fun isCrateRoot(file: VirtualFile): Boolean = findTargetForCrateRootFile(file) != null
 
-    fun withAdditionalPackages(additionalPackages: Collection<Pair<String, VirtualFile>>): CargoProjectDescription {
+    fun withAdditionalPackages(additionalPackages: Collection<Pair<String, VirtualFile>>, origin: PackageOrigin): CargoProjectDescription {
```

Comment:
Totally agree! I thought there was some bigger idea behind this function, but if it's that simple, it's better to do as you said.